flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch6/Elasticsearch6UpsertTableSink.java

```diff
@@ -189,6 +207,67 @@ protected ElasticsearchUpsertTableSinkBase copy(
 	// Helper classes
 	// --------------------------------------------------------------------------------------------
 
+	/**
+	 * This class implements {@link RestClientFactory}, used for es with authentication.
+	 */
+	static class AuthRestClientFactory implements RestClientFactory {
```

Comment:
Does it make sense to add userName and password as two optional fields to the `DefaultRestClientFactory` instead of introducing this class?